### DISCUSSION TITLE
Add Soundblaster Live! support

### DISCUSSION THREAD

**Discussion by shitsux420** [86Box/86Box#2795] [Ideas]
**Is your feature request related to a problem? Please describe.**
"Best" soundcard available currently is the Ensoniq AudioPCI, a quite similar card.

**Describe the solution you'd like**
Add support for the card. Has better drivers and DOS compatibility than Ensoniq's card.

**Describe alternatives you've considered**
Stick to Ensoniq AudioPCI.

**Additional context**
Maybe the "highest end" soundcard the emulator could/should support since the cut-off point for other features is around the same timeframe the card came out, eg. circa 1999. Actual 5.1 support is probably not needed.

**Comment by OBattler** [comment]
Converted to discussion per the policy to do that with feature requests before we decide whether or not to actually add the requested feature.

My question is - do we have any sort of documentation (datasheet or technical reference) for this card? From what I know, Creative stuff tends to be poorly documented.

**Comment by shitsux420** [comment]
I've been scouing the net for a few days for some docs for this card, but so far have found very little. I'll keep Googling.

**Reply by shitsux420** [reply]
I've been a bit busy lately, but been searching here and there and found 0 non bogus datasheets or technical manuals for the SB Live!. My final ideas have been to ask at vogons.org if maybe somebody has them stashed, but that forum is kinda slow these days, and maybe to ask straight from Creative or scour social media platforms for old Creative engineers who worked there at that time, but that will take time and energy I necessarily dont want to invest in alone.

**Reply by KoghaandSooga** [reply]
"but that forum is kinda slow these days," LOL. Tell me about it. Your more likely to get more activity in Kakariko Graveyard and maybe find a ghost........... in the daytime? (In game joke)

**Comment by Torinde** [comment]
Related #3651

**Reply by Torinde** [reply]
My simplistic view: SB Live! = Ensoniq AudioPCI + AWE64 (both already emulated, so should be feasible to implement?)

Live! and Audigy/LS and Audigy 2 Value/NX/ZS source code documentation (potentially partial). https://github.com/torvalds/linux/tree/master/sound/pci/emu10k1

**Reply by Torinde** [reply]
Illumos also has [drivers](https://illumos.org/hcl/): [code](https://github.com/search?q=repo%3Aillumos%2Fillumos-gate%20audigy&type=code)
Live! and Audigy/2/4
```
Creative Labs EMU10k1 [Sound Blaster Live! Series]
Driver: audioemu10k
PCI ID: 1102,2
Originating manifest: driver-audio-audioemu10k.p5m
Creative Labs EMU10k2/CA0100/CA0102/CA10200 [Sound Blaster Audigy Series]
Driver: audioemu10k
PCI ID: 1102,4
Originating manifest: driver-audio-audioemu10k.p5m
Creative Labs CA0108/CA10300 [Sound Blaster Audigy Series]
Driver: audioemu10k
PCI ID: 1102,8
Originating manifest: driver-audio-audioemu10k.p5m
```

**Reply by Torinde** [reply]
https://github.com/sbemu-x/sbemu-x/issues/12#issuecomment-1837526761 mentions multiple open source repositories with sound card drivers - covering Creative models among others: [OSS project](http://www.opensound.com/oss.html), [FreeBSD sound driver sources](https://cgit.freebsd.org/src/tree/sys/dev/sound/pci/hda), [ALSA drivers](https://alsa-project.org/), [IBM Audio Interface Library for 32-bit DOS (AIL/32) v.1.05](https://github.com/Wohlstand/ail32-sandbox)

SBEMU itself is a DOS driver for SB Live! (among others) based on mpxplay.

**Reply by Torinde** [reply]
Latest [Nuked-OPL3](https://github.com/nukeykt/Nuked-OPL3) supports stereo-extension, 4-ch output

**Comment by lazerode** [comment]
Creative still has the user guide for the Sound Blaster Live!. You can find it here:
https://files.creative.com/manualdn/Manuals/TSD/7557/Creative%20Labs%20SB%20Live%205_1%20Bulk%20English.pdf

**Reply by KoghaandSooga** [reply]
Thanks. The more documentation we have the better!
